cpp/tests/test_shuffler.cpp

```diff
@@ -173,8 +173,6 @@ void test_shuffler(
             sort_table(result), sort_table(expect_partitions[finished_partition])
         );
     }
-
-    shuffler.shutdown();
```

Comment:
IIRC, this was very much needed to ensure proper synchronization, maybe we'll need at least a `barrier()`, but let's see.

Reply:
Shuffler is shutdown where it was created or if its a class member, during the `TearDown` method.